src/tangerine/resources/assistant.py

```diff
@@ -323,6 +334,96 @@ def _log_interaction(
             log.exception("Failed to log interaction")
 
 
+class AssistantAdvancedChatApi(AssistantChatApi):
+    """
+    API for advanced assistant chat supporting multiple assistants,
+    external chunk injection, and model override.
+    """
+
+    def _convert_chunk_array_to_documents(self, chunks):
+        """
+        Converts an array of chunks into a list of Document objects.
+        """
+        return [
+            SearchResult(document=Document(page_content=chunk, metadata={}), score=1)
+            for chunk in chunks
+        ]
+
+    def post(self, _id=None):
+        assistant_names = request.json.get("assistants")
+        assistants = []
+
+        if not assistant_names:
+            return {"message": "assistant name(s) required"}, 400
+        try:
+            assistants = self._get_assistants(assistant_names)
+        except ValueError as err:
+            return {"message": str(err)}, 400
+
+        assistant_ids = [assistant.id for assistant in assistants]
+        question = request.json.get("query")
+        if not question:
+            return {"message": "query is required"}, 400
+        system_prompt = request.json.get("system_prompt", DEFAULT_SYSTEM_PROMPT)
+        session_uuid = request.json.get("sessionId", str(uuid.uuid4()))
+        stream = request.json.get("stream", "true") == "true"
+        previous_messages = request.json.get("prevMsgs")
+        interaction_id = request.json.get("interactionId", None)
+        client = request.json.get("client", "unknown")
+        model_name = request.json.get("model", "default")
+        model = MODELS.get(model_name)
+        if model is None:
+            return {"message": f"Unknown model: {model_name}"}, 400
+        embedding = embed_query(question)
+        chunks = request.json.get("chunks", None)
+        if chunks:
+            chunks = self._convert_chunk_array_to_documents(request.json.get("chunks"))
+        search_results = chunks or search_engine.search(assistant_ids, question, embedding)
```

Comment:
I think the ability to augment the chunks could be a good thing, merging them together, what do you think?

Reply:
I was asking if we could join both together?